src/field/vecops.cxx

```diff
@@ -175,14 +180,18 @@ const Field3D Div(const Vector3D &v, CELL_LOC outloc) {
 const Field2D Div(const Vector2D &v, const Field2D &f, CELL_LOC outloc) {
   TRACE("Div( Vector2D, Field2D )");
 
-  ASSERT1(v.getLocation() == f.getLocation());
-
-  Mesh *localmesh = f.getMesh();
-
   if (outloc == CELL_DEFAULT) {
     outloc = v.getLocation();
   }
 
+  ASSERT1(outloc != CELL_VSHIFT);
+
+  // The following probably belongs in the FDD? routines rather than here
+  // as we don't directly interact v and f. Note also no equivalent in Field3D
+  // version of this routine.
+  ASSERT1(outloc == f.getLocation());
```

Comment:
Could you say why the Field3D version doesn't have this?

Reply:
I don't know why -- it's just an observation to motivate why we might want to remove/move this!

Reply:
This is checked by indexFDD*, so we can remove it from here :+1: